clients/guests_app/ViewModel/OrderViewModel.tsx

```diff
@@ -0,0 +1,57 @@
+import { resolvePlugin } from "@babel/core";
+import { createOrder } from "../communication/requests";
+import { OrderModel } from "../Model/OrderModel";
+import {} from "../communication/requests"
+
+type Location = {
+    x: number;
+    y: number;
+};
+
+type Item = {
+    id: String
+    name: String 
+    time: Number
+}
+type LocalizationDetailsIDO = {
+
+}
+export type Order = {
+    id: OrderID;
+    items: Map<String,Number>;
+    status: OrderStatus;
+};
+
+type Arrived = boolean;
+type OrderID = string;
+type WaiterID = string;
+type OrderStatus = "a" | "B" // what status will we be on each system?
+
+
+export default class OrderViewModel{
+
+    private order_model = OrderModel.getInstance();
+
+    createOrder(items: Map<String,Number>):Promise<OrderID>
+    {
+        return createOrder(items)
+        .then((res) => {this.order_model.setOrder({id: res.data, items: items, status: "recieved"}); return res.data})
```

Comment:
I really like how you separated the responsibilities.
The view model is handling the login and calls to specific  requests as expected!

Minor detail here is that the function createOrder return a type of AxiosResponse<orderId>.
By doing that the networking component exposes its implementation to the ViewModel.

To keep think independent from each other it will be best the "extraction" of order id from the request will be at the requests file and the return type will be OrderID.
```suggestion
        .then((orderId) => {this.order_model.setOrder({id: orderId, items: items, status: "recieved"}); return orderId})
```